src/v/redpanda/admin/field_registry.cc

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "redpanda/admin/field_registry.h"
+
+#include "absl/strings/str_split.h"
+#include "src/v/redpanda/admin/tests/aip_filter_test_messages.proto.h"
+
+#include <string_view>
+#include <type_traits>
+
+namespace admin {
+
+template<RedpandaProtobufMessage T>
+field_accessor_info<T> protobuf_field_registry<T>::get_field_info(
+  const std::string& field_path) const {
+    auto field_numbers_opt = convert_field_path_to_numbers(field_path);
+    if (!field_numbers_opt) {
+        throw std::invalid_argument("Invalid field path: " + field_path);
+    }
+
+    T sample_instance;
+    auto field_opt = sample_instance.lookup_field(*field_numbers_opt);
+    if (!field_opt) {
+        throw std::invalid_argument("Field not found: " + field_path);
+    }
```

Comment:
this is not needed, because `convert_field_path_to_numbers` will never return a number list without this method also always returning a value